lib/std/json.zig

```diff
@@ -1483,7 +1483,22 @@ fn parseInternal(comptime T: type, token: Token, tokens: *TokenStream, options:
                                         parseFree(field.field_type, @field(r, field.name), options);
                                     }
                                 }
-                                @field(r, field.name) = try parse(field.field_type, tokens, options);
+                                var parsed_value = try parse(field.field_type, tokens, options);
+                                if (!field.is_comptime) {
+                                  @field(r, field.name) = parsed_value;
+                                } else {
+                                  // As we never set the field to the parsed value, make sure we free it.
+                                  defer parseFree(field.field_type, parsed_value, options);
+                                  // We have a comptime field. We need to check the value supplied in the
+                                  // json matches what the comptime value is.
+                                  if (field.default_value) |default| {
+                                    switch (@typeInfo(field.field_type)) {
+                                      .Bool, .Int, .Float => if (parsed_value != default) return error.ParsedValueNotDefault,
+                                      .Pointer => |pointerInfo| if (!mem.eql(pointerInfo.child, parsed_value, default)) return error.ParsedValueNotDefault,
```

Comment:
I don't think direction will work for e.g. pointer-to-pointer fields

Reply:
Are you thinking that
1. we need to avoid doing the comparison for pointer-to-pointer
2. or that there might be some situation where you define a field as a pointer-to-pointer that you actually want to parse json into (my imagination is failing me as to what an example of such would look like)?